pkg/scheduler/scheduler.go

```diff
@@ -450,6 +471,10 @@ func (s *Scheduler) forwardRequestToQuerier(querier schedulerpb.SchedulerForQuer
 	// Make sure to cancel request at the end to clean up resources.
 	defer s.cancelRequestAndRemoveFromPending(req.FrontendAddress, req.QueryID, "request complete")
 
+	queryComponentName := req.ExpectedQueryComponentName()
+	s.requestQueue.QueryComponentCapacity.IncrementForComponentName(queryComponentName)
```

Comment:
Calling `IncrementForComponentName` here introduces a race: we'll make a decision based on the current utilisation in `RequestQueue.trySendNextRequestForQuerier`, but only later update the utilisation here. In the meantime, we may have made more decisions for other requests without accounting for this query.

I think this has to be done synchronously as part of `RequestQueue`'s dispatcher loop.

Reply:
from slack discussion: will add comment for now, as the RequestQueue is not ready to utilize this

Reply:
moved